server/service/src/invoice/prescription/update/generate.rs

```diff
@@ -0,0 +1,107 @@
+use chrono::Utc;
+
+use repository::{InvoiceRow, InvoiceRowStatus, StockLineRow, StorageConnection};
+
+use crate::invoice::common::{
+    generate_batches_total_number_of_packs_update, InvoiceLineHasNoStockLine,
+};
+
+use super::{UpdatePrescription, UpdatePrescriptionError, UpdatePrescriptionStatus};
+
+pub(crate) struct GenerateResult {
+    pub(crate) batches_to_update: Option<Vec<StockLineRow>>,
+    pub(crate) update_invoice: InvoiceRow,
+}
+
+pub(crate) fn generate(
+    existing_invoice: InvoiceRow,
+    UpdatePrescription {
+        id: _,
+        status: input_status,
+        patient_id: input_patient_id,
+        clinician_id: input_clinician_id,
+        comment: input_comment,
+        colour: input_colour,
+    }: UpdatePrescription,
+    connection: &StorageConnection,
+) -> Result<GenerateResult, UpdatePrescriptionError> {
+    let should_update_batches_total_number_of_packs =
+        should_update_batches_total_number_of_packs(&existing_invoice, &input_status);
+    let mut update_invoice = existing_invoice.clone();
+
+    set_new_status_datetime(&mut update_invoice, &input_status);
+
+    update_invoice.name_id = input_patient_id.unwrap_or(update_invoice.name_id);
+    update_invoice.clinician_id = input_clinician_id.or(update_invoice.clinician_id);
+    update_invoice.comment = input_comment.or(update_invoice.comment);
+    update_invoice.colour = input_colour.or(update_invoice.colour);
+
+    if let Some(status) = input_status.clone() {
+        update_invoice.status = status.full_status().into()
+    }
+
+    let batches_to_update = if should_update_batches_total_number_of_packs {
```

Comment:
Only cause i wanted to do some coding, if statement is pretty clear already, just thought to show an alternative: https://github.com/openmsupply/open-msupply/compare/1951-Endpoint-Update-Prescription...1951-Endpoint-Update-Prescription(Stylistic-suggestion)?diff=unified